lib/src/hls/hls_parser/hls_playlist_parser.dart

```diff
@@ -622,8 +623,12 @@ class HlsPlaylistParser {
     required String pattern,
     required bool defaultValue,
   }) {
-    final List<Match> list = line.allMatches(pattern).toList();
-    return list.isEmpty ? defaultValue : list.first.pattern == booleanTrue;
+    final regExp = RegExp(pattern);
```

Comment:
Why did you change this code?

Reply:
Using String can't find the pattern, I don't know why, so I need to use Regex to find.

I leave a code that can be used in Dartcode to show that just don't work as expected.

```
void main() { 
  final line = '#EXT-X-MEDIA:TYPE=SUBTITLES,GROUP-ID="subs",NAME="English",DEFAULT=YES,AUTOSELECT=YES,FORCED=NO,LANGUAGE="en",URI="subtitles_en.m3u8"';
  final pattern = 'DEFAULT=(YES|NO)';
  
  print('Found with Regex first: ${useRegexToFind(line, pattern)}');
  print('Found with Regex From String: ${useRegexInList(line, pattern)}');  
}

bool useRegexToFind(String line, String patternString){
  final pattern = RegExp(patternString);  
  final List<Match> list = pattern.allMatches(line).toList();
  
  return list.isEmpty; //If true, dont find
}

bool useRegexInList(String line, String pattern){
  final List<Match> list = line.allMatches(pattern).toList();
  return list.isEmpty; //If true, dont find
}

```
I leave the results too.

![Captura de Tela 2021-09-13 às 09 13 59](https://user-images.githubusercontent.com/5496305/133081513-d3c87c2f-50f2-4550-9572-04494156bd95.png)